google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/RateLimitingServerStreamingCallable.java

```diff
@@ -0,0 +1,150 @@
+  private final static long minimumTimeBetweenUpdates = 60_000;
+
+  private RateLimiter limiter; // Capture .setRate() and pass in mocked RateLimiter
+  private RateLimitingStats stats;
+  private final ServerStreamingCallable<RequestT, ResponseT> innerCallable;
+
+  public RateLimitingServerStreamingCallable(
+      @Nonnull ServerStreamingCallable<RequestT, ResponseT> innerCallable) {
+    limiter =  RateLimiter.create(DEFAULT_QPS);
+    stats = new RateLimitingStats();
+    this.innerCallable = Preconditions.checkNotNull(
+        innerCallable, "Inner callable must be set");
+
+  }
+
+  public RateLimitingServerStreamingCallable(
+      @Nonnull ServerStreamingCallable<RequestT, ResponseT> innerCallable, RateLimitingStats stats) {
+    this.limiter =  RateLimiter.create(DEFAULT_QPS);
+    this.stats = stats;
+    this.innerCallable = Preconditions.checkNotNull(
+        innerCallable, "Inner callable must be set");
+
+  }
+  @Override
+  public void call(
+      RequestT request, ResponseObserver<ResponseT> responseObserver, ApiCallContext context) {
+    //if (context.getOption(Util.GRPC_METADATA) == null) { // get metadata?
+    //  context = context.withOption(Util.GRPC_METADATA, new GrpcResponseMetadata());
+    //}
+    //final GrpcResponseMetadata responseMetadata = context.getOption(Util.GRPC_METADATA);
+    //final ApiCallContext contextWithResponseMetadata = responseMetadata.addHandlers(context);
+
+    limiter.acquire();
+    CpuMetadataResponseObserver<ResponseT> innerObserver =
+        new CpuMetadataResponseObserver<>(responseObserver);
+
+    innerCallable.call(request, innerObserver, context);
+    //ApiFutures.addCallback(future, new CpuThrottlingUnaryCallback<>(responseMetadata), MoreExecutors.directExecutor());
+  }
+
+  private class CpuMetadataResponseObserver<ResponseT> extends SafeResponseObserver<ResponseT> {
+
+    private final ResponseObserver<ResponseT> outerObserver;
+    //private final GrpcResponseMetadata responseMetadata;
+
+    CpuMetadataResponseObserver(
+        ResponseObserver<ResponseT> observer) {
+      super(observer);
+
+      this.outerObserver = observer;
+      //this.responseMetadata = metadata;
+    }
+
+    @Override
+    protected void onStartImpl(final StreamController controller) {
+      outerObserver.onStart(controller);
+    }
+
+    @Override
+    protected void onResponseImpl(ResponseT response) { // response is right here...
+      System.out.println("On Response");
+      MutateRowsResponse mutateResponse = (MutateRowsResponse) response; // cast-safe?
+      ServerStats serverStats = mutateResponse.getServerStats();
+
+
+      System.out.println(serverStats.getCpuStatsCount());
+      // Where will I check the onError or onComplete?
+      System.out.println(mutateResponse.getEntriesList().toString()); // I don't know if I need to handle errors and retries differently?
+      // Weihan added the change to the proto
+      // What process do we need to get this proto?
```

Comment:
I'm not sure if we want to publish the protos just yet. So I think manually adding the generated classes here is good for now.